devito/types/basic.py

```diff
@@ -622,7 +622,7 @@ def __new__(cls, *args, **kwargs):
 
         if obj is not None:
             newobj = sympy.Function.__new__(cls, *args, **options)
-            newobj.__init_cached__(key)
+            newobj.__init_cached__(key, ignore=['_on_grid'])
```

Comment:
this **looks** hacky

Reply:
It's not, the same way we don;t cache on `indices` but don's see it since it's in sympy.Function. This is an attribute that is different for different subs/indices and should not be cached as the original function. And in generality, there is not reason ine would not want to do partial caching of the dict.

Reply:
changed, was making my life complicated, much easier to call new directly at subs.